src/view/gui/fMainWindow.cpp

```diff
@@ -252,9 +254,13 @@ fMainWindow::fMainWindow()
   m_tabWidget->setMinimumHeight(minheight);
   m_tabWidget->setMaximumHeight(m_tabWidget->minimumHeight());
 
-  m_toolTabdock = new QDockWidget();
+  m_toolTabdock = new DragAndDropSharingDockWidget(); // custom class to propagate drag-and-drop events to the main window
```

Comment:
Please make sure there is no memory leak.

Reply:
I just added a check and delete for this in fMainWindow's destructor.

Reply:
@AlexanderGetka-cbica  In Qt, QObjects and derivatives get destroyed when their parent does. In this particular case, it looks like the correct parent is fMainWindow itself. Do

```cpp
m_toolTabdock = new QDockWidget(this);
```

and then you don't have to worry about lifetime management.

(QMainWindow::addDockWidget() might set parent regardless, but whatever)

Reply:
@dboun I was aware of this and tried to set its parent directly to be fMainWindow. For some reason we couldn't figure out  yesterday, this causes the entire docking tool tab to be hidden on the gui, even if we explicitly call show() on it.

Reply:
I just discovered that setting the window flag for the dock widget to Qt::Window causes it to be hidden if it's a child of the main window. I set it to Qt::Subwindow and this works fine on my machine running Ubuntu, but the window flags have a platform-dependent effect so it's possible this ends up looking strange on mac/windows.